### PR TITLE
feat(DMS): add datasource rocketmq users

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add datasource rocketmq users
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccDmsRocketMQUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsRocketMQUser_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsRocketMQUser_basic
=== PAUSE TestAccDmsRocketMQUser_basic
=== CONT  TestAccDmsRocketMQUser_basic
--- PASS: TestAccDmsRocketMQUser_basic (747.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       747.489s

make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccDmsRocketMQUsers_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsRocketMQUsers_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsRocketMQUsers_basic
=== PAUSE TestAccDmsRocketMQUsers_basic
=== CONT  TestAccDmsRocketMQUsers_basic
--- PASS: TestAccDmsRocketMQUsers_basic (745.50s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       745.544s
```

### CODE REVIEW

File: docs/data-sources/dms_rocketmq_users.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+* `admin` - Indicates whether the user is an administrator.
+
+* `default_topic_perm` - Indicates the default topic permissions.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
+
+* `default_group_perm` - Indicates the default consumer group permissions.
+  Value options: **SUB**, **DENY**.
+
+* `topic_perms` - The list of the special topic permissions.
+  The [permission](#DMS_rocketmq_users_topic_perms) structure is documented below.
```
Comment (houpeng80) [inline docs/data-sources/dms_rocketmq_users.md]:
```suggestion
  The [topic_perms](#DMS_rocketmq_users_topic_perms) structure is documented below.
```

File: docs/data-sources/dms_rocketmq_users.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+<a name="DMS_rocketmq_users_topic_perms"></a>
+The `topic_perms` block supports:
+
+* `name` - Indicates the name of a topic.
+
+* `perm` - Indicates the permissions of the topic.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
+
+* `group_perms` - The list of the special consumer group permissions.
+  The [permission](#DMS_rocketmq_users_group_perms) structure is documented below.
```
Comment (houpeng80) [inline docs/data-sources/dms_rocketmq_users.md]:
```suggestion
  The [group_perms](#DMS_rocketmq_users_group_perms) structure is documented below.
```
move to line  71

File: huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rocketmq_users_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+output "default_topic_perm_filter_is_useful" {
+  value = length(data.huaweicloud_dms_rocketmq_users.default_topic_perm_filter.users) > 0 && alltrue(
+  [for v in data.huaweicloud_dms_rocketmq_users.default_topic_perm_filter.users[*].default_topic_perm : v == local.default_topic_perm]
+  )
+}
+
+data "huaweicloud_dms_rocketmq_users" "default_group_perm_filter" {
+	depends_on         = [huaweicloud_dms_rocketmq_user.test]
+	instance_id        = huaweicloud_dms_rocketmq_instance.test.id
+	default_group_perm = huaweicloud_dms_rocketmq_user.test.default_group_perm
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rocketmq_users_test.go]: remove the extra space, and fix other places

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+		if rawWhiteRemoteAddressOK && rawWhiteRemoteAddress != whiteRemoteAddress {
+			continue
+		}
+		if rawAdminOK {
+			if rawAdmin != admin {
+				continue
+			}
+		} else if admin {
+			continue
+		}
```
Comment (houpeng80) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go]:
```suggestion
		if (rawAdmin && !admin ) || (!rawAdmin && admin) {
			continue
		} 
```

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+	userJson := utils.PathSearch("users", resp, make([]interface{}, 0))
+	userArray := userJson.([]interface{})
+	if len(userArray) < 1 {
+		return nil
+	}
+	result := make([]interface{}, 0, len(userArray))
+
+	rawAccessKey, rawAccessKeyOK := d.GetOk("access_key")
+	rawWhiteRemoteAddress, rawWhiteRemoteAddressOK := d.GetOk("white_remote_address")
+	rawAdmin, rawAdminOK := d.GetOk("admin")
```
Comment (houpeng80) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go]:
```suggestion
	rawAdmin := d.Get("admin").(bool)
```

File: docs/data-sources/dms_rocketmq_users.md (hunk 1 above)
Comment (346010762) [reply docs/data-sources/dms_rocketmq_users.md]: done

File: docs/data-sources/dms_rocketmq_users.md (hunk 2 above)
Comment (346010762) [reply docs/data-sources/dms_rocketmq_users.md]: done

File: huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rocketmq_users_test.go (hunk 3 above)
Comment (346010762) [reply huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rocketmq_users_test.go]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go (hunk 4 above)
Comment (346010762) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go (hunk 5 above)
Comment (346010762) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_users.go]: done

**Pull request merged**